docs/tutorial/flask.rst

```diff
@@ -0,0 +1,530 @@
+
+.. code-block:: bash
+
+    mkdir flask-hello-world
+    cd flask-hello-world
+
+Create the Flask application
+============================
+
+Start by creating the "Hello, world" Flask application that will be
+used for this tutorial.
+
+Create a ``requirements.txt`` file, copy the following text into it
+and then save it:
+
+.. literalinclude:: code/flask/requirements.txt
+
+In the same directory, copy and save the following into a text file
+called ``app.py``:
+
+.. literalinclude:: code/flask/app.py
+    :language: python
+
+Run the Flask application locally
+=================================
+
+Install ``python3-venv`` and create a virtual environment:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:create-venv]
+    :end-before: [docs:create-venv-end]
+    :dedent: 2
+
+Now that we have a virtual environment with all the dependencies, let's
+run the Flask application to verify that it works:
+
+.. code-block:: bash
+
+    flask run -p 8000
+
+Test the Flask application by using ``curl`` to send a request to the root
+endpoint. You may need a new terminal for this; if you are using Multipass
+use ``multipass shell charm-dev`` to get another terminal:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:curl-flask]
+    :end-before: [docs:curl-flask-end]
+    :dedent: 2
+
+The Flask application should respond with ``Hello, world!``. The Flask
+application looks good, so we can stop for now using
+:kbd:`Ctrl` + :kbd:`C`.
+
+Pack the Flask application into a rock
+======================================
+
+First, we'll need a ``rockcraft.yaml`` file. Rockcraft will automate its
+creation and tailoring for a Flask application by using the
+``flask-framework`` profile:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:create-rockcraft-yaml]
+    :end-before: [docs:create-rockcraft-yaml-end]
+    :dedent: 2
+
+The ``rockcraft.yaml`` file will automatically be created and set the name
+based on your working directory. Open the file in a text editor and check
+that the ``name`` is ``flask-hello-world``. Ensure that ``platforms``
+includes the architecture of your host. For example, if your host uses the
+ARM architecture, include ``arm64`` in ``platforms``.
+
+.. note::
+
+    For this tutorial, we'll use the ``name`` "flask-hello-world" and assume
+    you are on the ``amd64`` platform. Check the architecture of your system
+    using ``dpkg --print-architecture``. Choosing a different name or
+    running on a different platform will influence the names of the files
+    generated by Rockcraft.
+
+Pack the rock:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:pack]
+    :end-before: [docs:pack-end]
+    :dedent: 2
+
+.. note::
+
+    Depending on your system and network, this step can take a couple of
+    minutes to finish.
```

Comment:
We could remove the note dressing here, too – information like step duration helps the reader develop a [tactile feel](https://diataxis.fr/tutorials/#target-the-feeling-of-doing) for the product.